geoh5py/data/data_type.py

```diff
@@ -324,7 +324,7 @@ def validate_data_type(workspace: Workspace, attribute_dict: dict):
                 ):
                     entity_type = {"primitive_type": "FLOAT"}
                 elif isinstance(values, np.ndarray) and (
-                    values.dtype in [np.uint32, np.int32]
+                    np.issubdtype(values.dtype, np.integer)
```

Comment:
I did a PR to change every type checking usin integer. Could yuou review it? :-)
https://github.com/MiraGeoscience/geoh5py/pull/528